Backend/src/database/database.py

```diff
@@ -43,9 +43,9 @@ def get_course(self, course_code: str, term: str) -> Course | None:
         '''
 
     @abstractmethod
-    def get_terms(self) -> List[str]:
+    def get_terms(self) -> List[str] | dict:
```

Comment:
Typically it is bad practice to have a method able to return multiple different types. Though right now our DynamoDB is not populated with the reading week dates so I'm not sure how we'd handle this. If we keep the code like this there will be an error if we ever switch back to DynamoDB (since it will not return a response with the expected format to the frontend). We should figure this out before merging.